### PR TITLE
add docs on using Kyverno CLI in a CI process; other fixes and updates

### LINKED ISSUE #484: Add documentation on how Kyverno CLI can be used in CICD pipelines
### Page link

_No response_

### Description

Add documentation on how Kyverno CLI can be used in CICD pipelines. 
Providing examples with GitHub Actions or Jenkins would be useful

### Expected behavior

Documentation is available for using Kyverno CLI in CICD pipelines. 

### Slack discussion

https://kubernetes.slack.com/archives/CLGR9BJU9/p1646846670880039

### LINKED ISSUE #491: Clarify any and all
### Page link

_No response_

### Description

Current documentation lacks a description of whether the any and all operators execute their checks in defined order?
Also it doesn't mention whether the checks are performed in a short-circuit manner so one can utilize the order to simplify any following checks.

https://en.wikipedia.org/wiki/Short-circuit_evaluation

### Slack discussion

_No response_

### LINKED ISSUE #561: [Enhancement] Add clusterrole aggregation docs
### Description

We now use clusterrole aggregation to configure controller permissions, we need to document the way users can bring their own permissions.

### Slack discussion

https://kubernetes.slack.com/archives/C032MM2CH7X/p1654628255764599

### LINKED ISSUE #569: [Enhancement] Document policy behaviors for `failurePolicy`
### Description

See: https://github.com/kyverno/kyverno/pull/4183 and https://github.com/kyverno/kyverno/issues/4168.

### Slack discussion

_No response_

### LINKED ISSUE #571: [Enhancement] Modify documentation to highlight that Imagepullsecrets accepts CSV of secrets
### Description

Description of the imagepullsecret deployment argument in https://kyverno.io/docs/writing-policies/verify-images/ (section Using private registries) seems to signify that the parameter takes one imagepullsecret.

Will be nice to have some description and/or example mentioning that it supports multiple imagepullsecrets as a CSV.

### Slack discussion

_No response_

### LINKED ISSUE #575: [Bug] Remove inconsistent statement about Namespace exclusion
### Page link

https://kyverno.io/docs/installation/#security-vs-operability

### Description

>The choices and their implications are therefore:
> 1. Do not exclude system Namespaces, including Kyverno’s, (not default) during installation resulting in a more secure-by-default posture but potentially requiring manual recovery steps in some outage scenarios. This is the default posture unless overridden.

Last sentence should be removed since exclusion of the Kyverno Namespace IS the default.

### Expected behavior

Last sentence is removed.

### Slack discussion

_No response_

### LINKED ISSUE #583: [Enhancement]  update hostNetwork: true section according to discussion
### Description

Discussion link : https://github.com/aws-samples/eks-blueprints-add-ons/pull/44#discussion_r898776198
page link : https://kyverno.io/docs/troubleshooting/#kyverno-fails-on-eks

### Slack discussion

_No response_

### LINKED ISSUE #588: [Enhancement] Chained context variables
### Description

Context variables which have a dependency order must be specified in a policy in that order. Ex., B uses A as an input to one of its JMESPath filters. A must be resolved first so B can be resolved. A must be placed above B in the rule definition.

### Slack discussion

_No response_

### LINKED ISSUE #589: [Bug] `path` not documented
### Page link

https://kyverno.io/docs/writing-policies/variables/#variables-from-container-images

### Description

We missed documenting the `path` field in variables from container images.

Ex,,

```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: pod-require-non-root-user
spec:
  rules:
    - name: require-non-root-user
      match:
        any:
        - resources:
            kinds:
              - Pod
      context:
      - name: contextVars
        configMap:
          name: kyverno-context-vars
          namespace: default
      mutate:
        foreach:
          - list: request.object.spec.containers
            preconditions:
              all:
                # skip images that are exempt (allowed to run as a root user);
                - key: "{{ images.containers.{{element.name}}.path }}"
                  operator: AnyNotIn
                  value: "{{contextVars.data.exemptRootUserImages}}"
            patchStrategicMerge:
              spec:
                containers:
                  - name: "{{ element.name }}"
                    securityContext:
                      runAsNonRoot: true
```

### Expected behavior

The `path` field shows up and an example value.

### Slack discussion

_No response_

### LINKED ISSUE #594: [Enhancement] Add docs for `policy.applyRules`
### Description

PR https://github.com/kyverno/kyverno/pull/4196.

>Allows users to control how rules in a policy are applied to a admission request or resource. By default all matching rules are applied. When applyRules: One is configured processing stops after the first rule is applied.

### Slack discussion

_No response_

### LINKED ISSUE #595: [Bug] State that auto-gen gets disabled upon heterogenous kinds
### Page link

https://main.kyverno.io/docs/writing-policies/autogen/

### Description

See https://github.com/kyverno/kyverno/issues/4306.

### Expected behavior

Add "Additionally, auto-generation of rules is skipped if the match and exclude blocks including additional resource types other than Pod."

### Slack discussion

_No response_

### LINKED ISSUE #596: [Enhancement] Add GitHub releases as an option for installing kyverno-cli
### Description

Almost every time, I use GitHub [releases](https://github.com/kyverno/kyverno/releases) page in order to install the kyverno CLI in my system. If we add this, then the users following the docs don't have to clone the repo and neither they need `go/make` to build the CLI. 

- We can add it here https://kyverno.io/docs/kyverno-cli/#building-and-installing-the-cli 

### Proposed docs

```
curl -LO https://github.com/kyverno/kyverno/releases/download/v1.7.2/kyverno-cli_v1.7.2_linux_x86_64.tar.gz
sudo tar -xvf kyverno-cli_v1.7.2_linux_x86_64.tar.gz -C /usr/local/bin
```
- With this we can also copy the kyverno CLI to kubectl-kyverno without the need to krew. 
```
sudo cp /usr/local/bin/kyverno /usr/local/bin/kubectl-kyverno
```
- This is mostly useful when we are creating new VM, GitHub codespaces, Gitpod & CI environments. 

**NOTE:**
I have only checked this on Linux instances, and I'm not sure about windows. On windows, we are not releasing the binary on the release page in `.exe` format so I'm not sure whether it will be the same for windows. 

### Slack discussion

_No response_

### LINKED ISSUE #597: [Bug] Incorrect field `mutateImages`
### Page link

https://main.kyverno.io/docs/writing-policies/verify-images/#verifying-image-signatures

### Description

Need to change `mutateImages` to `mutateDigest`. Eliminate duplicate "when".

### Expected behavior

Says `mutateDigest` and only one "when"

### Slack discussion

https://kubernetes.slack.com/archives/CLGR9BJU9/p1660230731207449

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: Chip Zoller <chipzoller@gmail.com>

## Related issue #

Closes #484
Fixes #597 
Closes #596
Closes #595 
Closes #594
Fixes #589 
Closes #588 
Closes #583 
Fixes #575 
Closes #571 
Closes #561 
Closes #569 
Closes #491 

## Proposed Changes

* Adds a new section on using the Kyverno CLI in a CI pipeline.
* Fixes typos and an inaccurate field reference in verifyImages
* Adds instructions for CLI manual installation
* Further clarifies auto-gen behavior
* Adds docs on new `spec.applyRules` field
* Add docs on missing `path` pre-defined container image var
* Clarifies context variable ordering
* Clarifies a troubleshooting note with EKS clusters using third-party CNI plug-ins
* Removes conflicting statement in the "Security vs Operability" section
* Expands info on `--imagePullSecrets` flag
* Adds and updates docs on the new ClusterRole aggregation changes
* Removes under construction notice from JMESPath page as this is decently mature by now
* Adds a note to the installation page warning that Kyverno must be installed in a dedicated Namespace
* Adds/updates docs on the changed behavior of `failurePolicy`
* Adds an Action which checks for broken links
* Adds phrase on preconditions and short circuiting

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

### CODE REVIEW

File: .github/workflows/check-links.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1.5.0
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      # - name: Create Issue From File
```
Comment (MarcelMue) [inline .github/workflows/check-links.yaml:21]: Is this here by mistake?

File: content/en/docs/Installation/_index.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ description: >
   Installation and configuration details for Kyverno using `Helm` or `kubectl`.
 ---
 
-Kyverno can be installed using Helm or deploying from the YAML manifests directly. When using either of these methods, there are no other steps required to get Kyverno up and running.
+Kyverno can be installed using Helm or deploying from the YAML manifests directly. When using either of these methods, there are no other steps required to get Kyverno up and running. Kyverno must always be installed in a dedicated Namespace; it must not be co-located with other applications in existing Namespaces including system-level Namespaces such as `kube-system`.
```
Comment (MarcelMue) [inline content/en/docs/Installation/_index.md:9]:
```suggestion
Kyverno can be installed using Helm or deploying from the YAML manifests directly. When using either of these methods, there are no other steps required to get Kyverno up and running.
Kyverno must always be installed in a dedicated Namespace; it must not be co-located with other applications in existing Namespaces including system-level Namespaces such as `kube-system`.
```

Let's use some extra newlines. They will render as a single line but it makes editing and diffs more readable.

File: content/en/docs/Testing Policies/_index.md [hunk 3]
```diff
@@ -3,9 +3,89 @@ title: "Testing Policies"
...
+  - workflow_dispatch
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Download Kyverno CLI
+        run: |
+          curl -sLO https://github.com/kyverno/kyverno/releases/download/v1.7.2/kyverno-cli_v1.7.2_linux_x86_64.tar.gz
```
Comment (MarcelMue) [inline content/en/docs/Testing Policies/_index.md:70]:
```suggestion
          export kyvernoversion = v1.7.2
          curl -sLO https://github.com/kyverno/kyverno/releases/download/${kyvernoversion}/kyverno-cli_${kyvernoversion}_linux_x86_64.tar.gz
```
Not entirely sure about the formatting on top of my head. The variable can then also be used in the line below. It makes it a bit easier to keep kyverno versions up to date.

**Review by MarcelMue — COMMENTED** [review]
Generally LGTM, just some small things

File: .github/workflows/check-links.yaml (hunk 1 above)
Comment (chipzoller) [reply .github/workflows/check-links.yaml:21]: No, it's commented out to remind me to revisit this later. The maintainer recently fixed a bug I filed regarding auto-opening of issues found by this action, I just haven't had time to verify.

File: content/en/docs/Testing Policies/_index.md (hunk 3 above)
Comment (chipzoller) [reply content/en/docs/Testing Policies/_index.md:70]: I'd rather move this to an env var at the top of the action which is easily done.

File: content/en/docs/Installation/_index.md (hunk 2 above)
Comment (chipzoller) [reply content/en/docs/Installation/_index.md]: Done.

**Review by MarcelMue — APPROVED** [review]
LGTM

**Pull request merged**